Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -84,15 +92,29 @@ class BTAnalyticsService: Equatable {
                 return
             }
 
-            let postParameters = self.createAnalyticsEvent(config: configuration, sessionID: self.apiClient.metadata.sessionID, event: event)
-            self.http?.post("v1/tracking/batch/events", parameters: postParameters) { _, _, _ in }
+            // Send analytics events immediately if no timer is needed for unit tests
+            if self.timerInterval == 0 {
+                self.sendQueuedAnalyticsEvents(configuration: configuration)
+            } else {
+                DispatchQueue.main.asyncAfter(deadline: .now() + self.timerInterval) { [weak self] in
```

Comment:
I changed the timer interval to 2 seconds and threw a print statement in this block. I would expect it to print to console every 2 seconds, but I'm not seeing that 👁️

Reply:
I think DispatchQueue.main.asyncAfter doesn't repeat a blocks execution numerous times, but just delays it by a certain period of time.

Since this is getting called every time we call `sendAnalyticsEvent()`, I think it's hiding the fact that this isn't working on a loop timer.

Reply:
That's a good call. I moved to `DispatchSourceTimer` here which _actually_ seems to repeat as expected when I add print statements: https://github.com/braintree/braintree_ios/pull/1295/commits/8a816a2b610e01fc12dcb5545493a50705da54d6.

Pros of this approach:
- Behaves as expected firing an event every however many seconds until it get's the the `BTAnalyticsService.events.isEmpty` else block and is canceled (can comment this line out to verify)
- Can be cancelled (vs running forever or until memory can be allocated)
- We don't need to manage run loops like we would with `Timer`

TODO: still need to address the thread-safety of the static array